### PR TITLE
feat: measure building domain

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/BuildingRegistry/Building/Commands/MeasureBuilding.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    using System;
+    using System.Collections.Generic;
+    using Be.Vlaanderen.Basisregisters.Generators.Guid;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Be.Vlaanderen.Basisregisters.Utilities;
+    using Datastructures;
+
+    public sealed class MeasureBuilding : IHasCommandProvenance
+    {
+        private static readonly Guid Namespace = new Guid("22e87692-b914-46a1-85f0-6c7e5a04fe1c");
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Commands/MeasureBuilding.cs:12]: Guid is not unique.

File: src/BuildingRegistry/Building/Commands/MeasureBuilding.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            Provenance provenance)
+        {
+            BuildingPersistentLocalId = buildingPersistentLocalId;
+            Geometry = geometry;
+            BuildingGrbData = buildingGrbData;
+            Provenance = provenance;
+        }
+
+        public Guid CreateCommandId()
+            => Deterministic.Create(Namespace, $"RealizeAndMeasureUnplannedBuilding-{ToString()}");
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Commands/MeasureBuilding.cs:35]: Change to class name.

File: src/BuildingRegistry/Building/Events/BuildingWasMeasured.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Newtonsoft.Json;
+
+    [EventTags(EventTag.For.Sync, EventTag.For.Edit, Tag.Building)]
+    [EventName(EventName)]
+    [EventDescription("Het gebouw werd ingemeten.")]
+    public sealed class BuildingWasMeasured : IBuildingEvent
+    {
+        public const string EventName = "BuildingWasMeasure"; // BE CAREFUL CHANGING THIS!!
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Events/BuildingWasMeasured.cs:16]: You forgot a "d" at the end.

File: src/BuildingRegistry/Building/Building_State.cs [hunk 4]
```diff
@@ -218,6 +219,16 @@ private void When(UnplannedBuildingWasRealizedAndMeasured @event)
             _lastEvent = @event;
         }
 
+        private void When(BuildingWasMeasured @event)
+        {
+            BuildingPersistentLocalId = new BuildingPersistentLocalId(@event.BuildingPersistentLocalId);
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building_State.cs:224]: You don't have to set the BuildingPersistentLocalId.

File: src/BuildingRegistry/Building/Building_Grb.cs [hunk 5]
```diff
@@ -52,5 +52,26 @@ public void Demolish(BuildingGrbData grbData)
...
+            if (BuildingGeometry.Method != BuildingGeometryMethod.MeasuredByGrb)
+            {
+                throw new BuildingHasInvalidGeometryMethodException();
+            }
+
+            GuardValidStatusses(BuildingStatus.Realized);
+
+            var geometry = WKBReaderFactory.Create().Read(extendedWkbGeometry);
+
+            GuardPolygon(geometry);
```
Comment (jvandaal) [inline src/BuildingRegistry/Building/Building_Grb.cs:71]: Can we move this validation up? I would validate incoming parameters first.

**Review by ArneD — APPROVED** [review]

**Pull request merged**